frontend/ui/uimanager.lua

```diff
@@ -189,7 +189,10 @@ function UIManager:show(widget, refreshtype, refreshregion, x, y)
     for i = #self._window_stack, 0, -1 do
         local top_window = self._window_stack[i]
         -- skip modal window
-        if not top_window or not top_window.widget.modal then
+        if widget.modal then
```

Comment:
I could just stick this in line with the other ors couldn't I. Silly me. I was making it too complicated. :-p